src/presign/participant.rs

```diff
@@ -186,10 +228,92 @@ impl PresignParticipant {
             ));
         }
 
-        Ok(ret_messages)
+        let mut broadcast_messages = self.broadcast(
+            rng,
+            &MessageType::Presign(PresignMessageType::RoundOneBroadcast),
+            serialize!(&r1_public_broadcast)?,
+            message.id(),
+            "PresignR1Ciphertexts",
+        )?;
+        out_messages.append(&mut broadcast_messages);
+
+        // Additionally, handle any round 1 messages which may have been received too early
+        let retrieved_messages = self.fetch_messages(
+            MessageType::Presign(PresignMessageType::RoundOne),
+            message.id(),
+        )?;
+        let mut presign_record_option = None;
+        for msg in retrieved_messages.iter() {
+            let (pr_option, mut r2_msg) = self.handle_round_one_msg(rng, msg, main_storage)?;
+            out_messages.append(&mut r2_msg);
+            presign_record_option = pr_option;
```

Comment:
question: What happens here if there is more than one retrieved message? Won't this get overwritten?

Reply:
Good call, yes, it would get overridden. I think this technically isn't a problem in normal operation, as only the final message you process could possibly return a `pr_option `that isn't `None`. But if there's extra/duplicate messages for some reason, it could mess with things